lib/spec/staticcacher.sw-spec.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+importScripts('/base/lib/staticcacher.js');
+importScripts('/base/lib/spec/dummies/cachehelper.js');
+
+describe('Static cacher', function () {
+
+  describe('The constructor', function () {
+
+    it('should accept an array of files', function () {
+      var fileList = ['file1', 'file2'];
+      var cacher = new StaticCacher(fileList);
+      expect(cacher.files).to.deep.equal(fileList);
+    });
+
+    it('should throw an error if no argument is provided', function () {
+      expect(function() { new StaticCacher() }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw an error if the argument is not an array', function () {
+      expect(function() { new StaticCacher('invalidString') }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw if the array is empty', function () {
+      expect(function() { new StaticCacher([]) }).to.throw(Error, 'Invalid file list');
+    });
+  });
+
+  describe('onInstall()', function () {
+    var files = ['file1', 'file2'];
+    var cacher = new StaticCacher(files);
+
+    var dummyCache = 'cache';
+    var getDefaultCacheStub = sinon.stub(CacheHelper, 'getDefaultCache').returns(Promise.resolve(dummyCache));
+    var addAllStub = sinon.stub(CacheHelper, 'addAll').returns(Promise.resolve());
+
+    afterEach(function() {
+      getDefaultCacheStub.reset();
+      addAllStub.reset();
+    });
+
+    it('should return a promise', function () {
+      return cacher.onInstall().then(function() {
```

Comment:
Notice you're testing it should return a promise, not the promise is being fulfilled. So you test should check _thenability_ of the returned value:

```js
var returned = cacher.onInstall();
expect(returned.then).to.be.a('function');
```